app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Formatters.scala

```diff
@@ -24,14 +24,21 @@ import scala.util.control.Exception.nonFatalCatch
 
 trait Formatters {
 
-  private[mappings] def stringFormatter(errorKey: String, args: Seq[String] = Seq.empty): Formatter[String] =
+  private[mappings] def stringFormatter(
+    errorKey: String,
+    args: Seq[String] = Seq.empty,
+    sanitise: String => String
+  ): Formatter[String] =
     new Formatter[String] {
 
       override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], String] =
         data.get(key) match {
-          case None                      => Left(Seq(FormError(key, errorKey, args)))
-          case Some(s) if s.trim.isEmpty => Left(Seq(FormError(key, errorKey, args)))
-          case Some(s)                   => Right(s)
+          case None    => Left(Seq(FormError(key, errorKey, args)))
+          case Some(s) =>
+            sanitise(s) match {
+              case "" => Left(Seq(FormError(key, errorKey, args)))
```

Comment:
you can use `case _.isEmpty`, using "" is discouraged in FP

Reply:
Ok Used `case ss if ss.isEmpty` as case `_.IsEmpty` didn't work!